mimic/rest/maas_api.py

```diff
@@ -30,6 +30,16 @@
 Request.defaultContentType = 'application/json'
 
 
+def _find_index_of_id(xs, val):
```

Comment:
OK, the difference is clear to me now.  Sorry I didn't catch that before.  I think I think this is probably fine, but let me propose an alternative.

`index` is doing an `==` comparison, which means that it's calling a method, `__eq__`, on each list item, with what it was passed.  Given Python's rules for equality comparison, this will actually call `__eq__` on the *left*-hand-side argument if the right-hand side doesn't recognize its type.

So, you can implement this a different way, supplying your own comparison data structure but using the built-in implementation of `index`.  Consider this example:

```python
class MatchesID(object):
    def __init__(self, id):
        self.id = id

    def __eq__(self, other):
        return other['id'] == self.id


values = [
    {"id": 3, "other": 4},
    {"id": 4, "other": 81},
    {"id": 5, "other": 24},
    {"id": 6, "other": 4},
    {"id": 3, "other": 9},
    {"id": 3, "other": 12},
    {"id": 6, "other": 14},
]

idx = (values.index(MatchesID(5)))
print(values[idx])
```

this will print out

```python
{'other': 24, 'id': 5}
```